test/e2e/ownerreference_helpers.go

```diff
@@ -224,3 +226,30 @@ func clusterResourceSetBindingIsOnlyOwnedByClusterResourceSets(
 ) error {
 	return framework.HasExactOwners(dedupeOwners(gotOwners), clusterResourceSetOwner)
 }
+
+// filterClusterObjectsWithNameFilterIgnoreClusterAutoscaler filters out all objects that are not part of the workload
+// cluster hierarchy as well as cluster autoscaler objects because they are not owned by any cluster object.
+func filterClusterObjectsWithNameFilterIgnoreClusterAutoscaler(
+	clusterName string,
+) clusterctlcluster.GetOwnerGraphFilterFunction {
+	return func(u unstructured.Unstructured) bool {
+		if !clusterctlcluster.FilterClusterObjectsWithNameFilter(
+			clusterName,
+		)(
+			u,
+		) {
+			return false
+		}
```

Comment:
Am I reading this incorrectly? I don't understand the syntax. 
```
		if !clusterctlcluster.FilterClusterObjectsWithNameFilter(clusterName)(u) {
			return false
		}
```

Reply:
Yes, sorry that's just bad formatting by golines 🤦 Fixed that now. Basically filter out anything without the cluster name in the resource name - this is the standard filtering that we adopted before this change.